src/palace/manager/sqlalchemy/model/licensing.py

```diff
@@ -1524,18 +1516,14 @@ def set(
             dirty = True
 
         if dirty:
-            # TODO: We need to explicitly commit here so that
-            # LicensePool.delivery_mechanisms gets updated. It would be
-            # better if we didn't have to do this, but I haven't been able
-            # to get LicensePool.delivery_mechanisms to notice that it's
-            # out of date.
-            if autocommit:
-                _db.commit()
```

Comment:
This commit buried deep in the code is offensive.  On the plus side,  I appreciate the TODO note.